src/main/java/sorald/processor/StringLiteralInsideEqualsProcessor.java

```diff
@@ -0,0 +1,77 @@
+package sorald.processor;
+
+import sorald.annotations.ProcessorAnnotation;
+import spoon.reflect.code.BinaryOperatorKind;
+import spoon.reflect.code.CtBinaryOperator;
+import spoon.reflect.code.CtExpression;
+import spoon.reflect.code.CtInvocation;
+import spoon.reflect.declaration.CtElement;
+import spoon.reflect.declaration.CtMethod;
+import spoon.reflect.declaration.CtType;
+import spoon.reflect.reference.CtExecutableReference;
+
+@ProcessorAnnotation(
+        key = 1132,
+        description =
+                "Strings literals should be placed on the left side when checking for equality")
+public class StringLiteralInsideEqualsProcessor extends SoraldAbstractProcessor<CtInvocation<?>> {
+
+    @Override
+    protected void repairInternal(CtInvocation<?> element) {
+        if (element.getExecutable().getSignature().equals("equals(java.lang.Object)")
+                || element.getExecutable()
+                        .getSignature()
+                        .equals("equalsIgnoreCase(java.lang.String)")) {
```

Comment:
This is not the right place to check that the element is amenable for repair, that should be done in `canRepairInternal`. An element that appears in `repairInternal` _must_ be repaired.

I would just remove this check entirely as there's no test to demonstrate that it's necessary, and so I'm not convinced that it is necessary.

Reply:
Changed.